lib/phoenix/endpoint/cowboy_handler.ex

```diff
@@ -114,4 +116,23 @@ defmodule Phoenix.Endpoint.CowboyHandler do
     addr_str = :inet.ntoa(addr)
     "Running #{inspect endpoint} with Cowboy using #{scheme}://#{addr_str}:#{port}"
   end
+
+  defp log_on_invalid_cowboy_version(endpoint) do
+    case Application.spec(:cowboy, :vsn) do
+      [?1 | _] -> nil
+      version ->
+        Logger.warn("""
+        This handler is for Cowboy 1.x and you are using Cowboy #{version}
```

Comment:
"This handler" is not clear. Probably something like this:

```
You have specified Phoenix.Handlers.Cowboy for cowboy v1.x in the
:handler configuration of your Phoenix endpoint #{inspect endpoint}
but your mix.exs has fetched cowboy v2.x.
```

The rest of the message is perfect.